pynicotine/gtkgui/chatrooms.py

```diff
@@ -205,12 +206,30 @@ def clear_notifications(self):
                 self.unhighlight_room(room)
                 break
 
-    def show_room(self, room):
+    def show_room(self, room, is_private=False, switch_page=True, remembered=False):
 
-        page = self.pages.get(room)
+        if room not in self.pages:
+            self.pages[room] = tab = ChatRoom(self, room, is_private=is_private)
+            is_global = (room == core.chatrooms.GLOBAL_ROOM_NAME)
 
-        if page is not None:
-            self.set_current_page(page.container)
+            if is_global:
+                tab_position = 0
+            elif remembered:
+                tab_position = -1
+            else:
+                tab_position = 1
```

Comment:
This is not the correct tab_position unless the Public global room tab is open.

Although, the bug existed before this PR.